cmd/grpcurl/grpcurl.go

```diff
@@ -290,6 +306,21 @@ func main() {
 	if *plaintext && *key != "" {
 		fail(nil, "The -plaintext and -key arguments are mutually exclusive.")
 	}
+	if *usealts && *insecure {
+		fail(nil, "The -alts and -insecure arguments are mutually exclusive.")
+	}
+	if *usealts && *cert != "" {
+		fail(nil, "The -alts and -cert arguments are mutually exclusive.")
+	}
+	if *usealts && *key != "" {
+		fail(nil, "The -alts and -key arguments are mutually exclusive.")
+	}
+	if *altsHandshakerServiceAddress != "" && !*usealts {
+		fail(nil, "The -alts-handshaker-service argument must be used with the -alts argument.")
+	}
+	if len(altsTargetServiceAccounts) > 0 && !*usealts {
+		fail(nil, "The -alts-target-service-account argument must be used with the -alts argument.")
+	}
```

Comment:
This is kind of a combinatorical mess.  Perhaps early up we could assign `usetls := !*plaintext && !*usealts`?

Then you could collapse cases like `*plaintext && *cert` and `*usealts && *cert` into something like `if *cert && !usetls { ... }`